### PR TITLE
GH-43169: [Swift] Add StructArray to ArrowReader

### PR DESCRIPTION
**Pull request by abandy**
### Rationale for this change
Structs have been added for Swift but currently the ArrowReader does not support them.  This PR adds the ArrowReader support

### What changes are included in this PR?
Adding StructArray to ArrowReader

### Are these changes tested?
The next PR for the ArrowWriter will include a test for reading and writing Structs.

* GitHub Issue: #43169

### CODE REVIEW

File: swift/Arrow/Sources/Arrow/ArrowReader.swift [hunk 1]
```diff
@@ -54,49 +86,104 @@ public class ArrowReader {
...
+    private func loadPrimitiveData(
+        _ loadInfo: DataLoadInfo,
+        field: org_apache_arrow_flatbuf_Field)
+    -> Result<ArrowArrayHolder, ArrowError> {
+        guard let node = loadInfo.batchData.nextNode() else {
+            return .failure(.invalid("Node not found"))
+        }
+
+        guard let nullBuffer = loadInfo.batchData.nextBuffer() else {
+            return .failure(.invalid("Buffer not found"))
```
Comment (kou) [inline swift/Arrow/Sources/Arrow/ArrowReader.swift:128]: How about adding more information which buffer type isn't found?

```suggestion
            return .failure(.invalid("Null buffer not found"))
```

File: swift/Arrow/Sources/Arrow/ArrowReader.swift [hunk 2]
```diff
@@ -54,49 +86,104 @@ public class ArrowReader {
...
+        }
+
+        guard let valueBuffer = loadInfo.batchData.nextBuffer() else {
+            return .failure(.invalid("Buffer not found"))
```
Comment (kou) [inline swift/Arrow/Sources/Arrow/ArrowReader.swift:132]:
```suggestion
            return .failure(.invalid("Value buffer not found"))
```

File: swift/Arrow/Sources/Arrow/ArrowReader.swift [hunk 3]
```diff
@@ -54,49 +86,104 @@ public class ArrowReader {
...
+    private func loadVariableData(
+        _ loadInfo: DataLoadInfo,
+        field: org_apache_arrow_flatbuf_Field)
+    -> Result<ArrowArrayHolder, ArrowError> {
+        guard let node = loadInfo.batchData.nextNode() else {
+            return .failure(.invalid("Node not found"))
+        }
+
+        guard let nullBuffer = loadInfo.batchData.nextBuffer() else {
+            return .failure(.invalid("Buffer not found"))
```
Comment (kou) [inline swift/Arrow/Sources/Arrow/ArrowReader.swift:154]: ditto.

**Review by kou — APPROVED** [review]
+1

**Comment by abandy** [comment]
@kou I hope all is well.  I think I have resolved the comments.  Please review when you get a chance.  Thanks again!!

**Pull request merged**